### PR TITLE
shared-mime-info: split outputs

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Mic92 — APPROVED** [review]
could not find any usage where it tried to access dev or man output in nixpkgs

**Comment by Mic92** [comment]
Mhm is it really worth it to split the package for one manpage?

```
╰─ % wc -c ./result/share/man/man1/update-mime-database.1.gz
780 ./result/share/man/man1/update-mime-database.1.gz
```

**Comment by Mic92** [comment]
I could not find dev output

**Comment by SuperSandro2000** [comment]
That is the only dev output next to mpv I have in my home-manager profile. 

```
 result-dev ⇒ /nix/store/wi8493arxarx96spsmz4ra9br06g5z8s-shared-mime-info-2.1-dev
├──  nix-support
│  └──  propagated-build-inputs
└──  share
└──  pkgconfig
└──  shared-mime-info.pc
 result-man ⇒ /nix/store/0pf295vcg2fvamqlc1m981ph38qd0v1j-shared-mime-info-2.1-man
└──  share
└──  man
└──  man1
└──  update-mime-database.1.gz
```

File: pkgs/data/misc/shared-mime-info/default.nix [hunk 1]
```diff
@@ -53,6 +53,8 @@ stdenv.mkDerivation rec {
     "-Dupdate-mimedb=true"
   ];
 
+  outputs = [ "out" "dev" "man" ];
```
Comment (jtojnar) [inline pkgs/data/misc/shared-mime-info/default.nix:56]: Conventionally, it goes after pname/version

**Comment by SuperSandro2000** [comment]
I tried building ``plasma5Packages.kio`` which failed at building gpgme because some none utf-8 got send over the remote protocol. Should be fixed in the next nixUnstable version.

**Pull request merged**